homeassistant/components/sensor/teksavvy.py

```diff
@@ -0,0 +1,168 @@
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_TOTAL_BANDWIDTH): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Setup the sensor platform."""
+    websession = async_get_clientsession(hass)
+    apikey = config.get(CONF_API_KEY)
+    bandwidthcapstr = config.get(CONF_TOTAL_BANDWIDTH)
+    try:
+        bandwidthcap = int(bandwidthcapstr)
+        ts_data = TekSavvyData(hass.loop, websession, apikey, bandwidthcap)
+        ret = yield from ts_data.async_update()
+    except ValueError as error:
+        _LOGGER.error("Failed conversion %s %s", CONF_TOTAL_BANDWIDTH, error)
+    if ret is False:
+        return
+
+    name = config.get(CONF_NAME)
+    sensors = []
+    for variable in config[CONF_MONITORED_VARIABLES]:
+        sensors.append(TekSavvySensor(ts_data, variable, name))
+    async_add_devices(sensors, True)
+
+
+class TekSavvySensor(Entity):
+    """TekSavvy Bandwidth sensor."""
+
+    def __init__(self, teksavvydata, sensor_type, name):
+        """Initialize the sensor."""
+        self.client_name = name
+        self.type = sensor_type
+        self._name = SENSOR_TYPES[sensor_type][0]
+        self._unit_of_measurement = SENSOR_TYPES[sensor_type][1]
+        self._icon = SENSOR_TYPES[sensor_type][2]
+        self.teksavvydata = teksavvydata
+        self._state = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return '{} {}'.format(self.client_name, self._name)
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        return self._unit_of_measurement
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return self._icon
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Get the latest data from TekSavvy and update the state."""
+        yield from self.teksavvydata.async_update()
+        if self.type in self.teksavvydata.data:
+            self._state = round(self.teksavvydata.data[self.type], 2)
+
+
+class TekSavvyData(object):
+    """Get data from TekSavvy API."""
+
+    def __init__(self, loop, websession, api_key, bandwidth_cap):
+        """Initialize the data object."""
+        self.loop = loop
+        self.websession = websession
+        self.api_key = api_key
+        self.bandwidth_cap = bandwidth_cap
+        self.data = dict()
+        self.data["limit"] = self.bandwidth_cap
+
+    @asyncio.coroutine
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def async_update(self):
+        """Get the TekSavvy bandwidth data from the web service."""
+        headers = {"TekSavvy-APIKey": self.api_key}
+        _LOGGER.info("Updaing TekSavvy data")
```

Comment:
Make this `_LOGGER.debug`. Less verbose.

Reply:
Done.